scarb/src/ops/resolve.rs

```diff
@@ -301,21 +301,23 @@ pub fn resolve_workspace_with_opts(
                 );
             }
 
-            let source_map = SourceMap::preloaded(ws.members(), ws.config());
-            let cached = RegistryCache::new(&source_map);
-            let patched = RegistryPatcher::new(&cached, &patch_map);
-
             let members_summaries = ws
                 .members()
                 .map(|pkg| pkg.manifest.summary.clone())
                 .collect::<Vec<_>>();
 
-            let lockfile: Lockfile = if opts.update {
-                Lockfile::new([])
+            let (lockfile, yanked_whitelist) = if opts.update {
+                (Lockfile::new([]), HashSet::new())
```

Comment:
👍 
We can doc somewhere that update will invalidate usage of any yanked versions (e.g. https://docs.swmansion.com/scarb/docs/reference/lockfile.html#updating-locked-versions)